tests/unit/deposit/test_deposit_api.py

```diff
@@ -74,3 +77,115 @@ def test_deposit_versioning_draft_child_unlinking_bug(
     # the edited draft
     assert pv.draft_child_deposit
     assert pv.draft_child
+
+
+def test_deposit_with_custom_field(
+    mocker, api, api_client, db, es, locations, users, write_token,
+        license_record, minimal_deposit, communities):
+    """Test deposit with custom field publishing."""
+    auth = write_token['auth_header']
+    headers = [
+        ('Content-Type', 'application/json'),
+        ('Accept', 'application/json')
+    ]
+    auth_headers = headers + auth
+
+    # Get deposit URL
+    with api.test_request_context():
+        deposit_url = url_for('invenio_deposit_rest.depid_list')
+
+    # Test wrong community
+    minimal_deposit['metadata']['custom'] = {
+        "not-custom-metadata-community": {
+            "family": {"value": "Felidae",
+                       "uri": "https://en.wikipedia.org/wiki/Felidae"},
+            "genus": {"value": "Felis",
+                      "uri": "https://en.wikipedia.org/wiki/Felis"},
+            "behavior": {"value": "Plays with yarn, sleeps in cardboard box."}
+            }
+    }
+    response = api_client.post(
+        deposit_url, data=json.dumps(minimal_deposit), headers=auth_headers)
+    assert response.json['errors'] == [{
+        'field': 'metadata.custom',
+        'message': 'The community: not-custom-metadata-community is'
+                   ' not supporting custom metadata'
+            }]
+
+    # Test wrong scheme
+    minimal_deposit['metadata']['custom'] = {
+        "custom-metadata-community": {
+            "non-existent-scheme": {
+                "value": "Felidae",
+                "uri": "https://en.wikipedia.org/wiki/Felidae"
+                }
+            }
+    }
+    response = api_client.post(
+        deposit_url, data=json.dumps(minimal_deposit), headers=auth_headers)
```

Comment:
![](https://i.imgur.com/vrsLBmp.gif?1)
```suggestion
        deposit_url, json=minimal_deposit, headers=auth_headers)
```